dbms/src/DataStreams/MergingAndConvertingBlockInputStream.h

```diff
@@ -141,31 +156,45 @@ class MergingAndConvertingBlockInputStream : public IProfilingBlockInputStream
                 if (parallel_merge_data->exception)
                     std::rethrow_exception(parallel_merge_data->exception);
 
-                auto it = parallel_merge_data->ready_blocks.find(current_bucket_num);
-                if (it != parallel_merge_data->ready_blocks.end())
+
+                if (!parallel_merge_data->ready_blocks.empty())
                 {
-                    ++current_bucket_num;
-                    scheduleThreadForNextBucket();
+                    size_t finished_thread_num = parallel_merge_data->ready_blocks.size();
+
+                    for (size_t i = 0; i < finished_thread_num; ++i)
+                    {
+                        scheduleThreadForNextBucket();
+                    }
 
-                    if (it->second)
+                    for (auto & ready_block : parallel_merge_data->ready_blocks)
                     {
-                        res.swap(it->second);
-                        break;
+                        current_bucket_num++;
+                        if (!ready_block.second.empty())
+                        {
+                            two_level_blocks.splice(two_level_blocks.end(), std::move(ready_block.second), ready_block.second.begin(), ready_block.second.end());
```

Comment:
Looks like `ready_block.first` is not used? Please double check that
1. Is it ok to not use `ready_block.first`
2. If it is safe to not use `ready_block.first`, than can we just change the type of `ready_blocks` to a vector instead of a map

Reply:
done.